### ISSUE TITLE
sendEmailAlert sometimes fires too early

### ISSUE THREAD

**Issue by pbchase** [vanderbilt-redcap/email-alerts-module#12]
The sendEmailAlert sometimes fires early denying other modules the chance to write data need for rules. We encountered this problem when email_alerts was configured to fire on a form wherein another module was writing a field that email_alerts need for its rules.  Email_alerts does use the `delayModuleExecution` function to allow it to run later than other modules, but it calls it _after_ the rules are tested. This caused email rules to fail before the other module wrote the data that would satisfy them. 

We have a fix for this problem in a pull request I'll submit shortly.

**Comment by pbchase** [comment]
PR #13 provides a fix for this issue.

**Comment by jasonlones** [comment]
Thanks Philip!  I am hoping this will resolve an issue we've had with Email Alerts since around 2.0.2.  When using the Save Survey to PDF ExMod in conjunction with Email Alerts, Email Alerts would fire before the Save Survey module finished.

**Comment by pbchase** [comment]
Yeah, that sounds like it would be fixed by this PR. Glad I could help.

**Comment by jasonlones** [comment]
This doesn't seem to have helped in our case.  We are still experiencing 'EmailAlerts' firing before 'SaveSurvey'.
